### DISCUSSION TITLE
How to prevent myself from using vpn?

### DISCUSSION THREAD

**Discussion by mtrazzi** [SelfControlApp/selfcontrol#684] [Q&A]
I notice that the only way to bypass the blocker is by using a vpn. The general workflow is that I type something like "vpn" in a google search and download the chrome extension. How do I do the regex in my blocked list to block such kind of searchs? I tried "\*vpn\*" or "*vpn" I think and it didn't work. Any tips?

**Comment by mtrazzi** [comment]
adding `chrome.google.com` and `addons.mozilla.org` to the blocklist worked

**Reply by imflash217** [reply]
what about MacOS VPN apps? How de we block those?

**Comment by Craig-Stowers** [comment]
weird.. i use ProtonVPN and SelfControl seems to block everything fine. Maybe the browser level VPN's work a bit different...
